build.sh

```diff
@@ -1,85 +1,105 @@
-#!/bin/bash
+#!/bin/sh
 
 # Stop on first error
 set -e
 
 #SEQ=!!!!! ENTER YOUR FAVORITE Y4M HERE !!!!!
 
-IS_RELEASE=1
+unset IS_RELEASE
 
 for arg in "$@"; do
-  shift
-  case "$arg" in
-    "--debug") IS_RELEASE=0 ;;
-    *)        set -- "$@" "$arg"
+  case $arg in
+  --debug) unset IS_RELEASE ;; # Uneeded, but explicit
+  --release) IS_RELEASE=true ;;
+  *) continue ;;
   esac
 done
 
-if [[ -z "${SEQ}" ]]; then
+if [ -z "$SEQ" ]; then
   SEQ=nyan.y4m
-  SEQ10=nyan10.y4m
-  SEQ12=nyan12.y4m
-
-  wget -nc https://mf4.xiph.org/~ltrudeau/videos/nyan.y4m
+  #SEQ10=nyan10.y4m
+  #SEQ12=nyan12.y4m
+
+  if type wget > /dev/null 2>&1; then
+    wget -nc https://mf4.xiph.org/~ltrudeau/videos/nyan.y4m
+  elif type curl > /dev/null 2>&1; then
+    curl -fkLs https://mf4.xiph.org/~ltrudeau/videos/nyan.y4m -o nyan.y4m
+  fi
   #wget -nc https://people.xiph.org/~tdaede/nyan10.y4m
   #wget -nc https://people.xiph.org/~tdaede/nyan12.y4m
 fi
 
-
 if [ ! -f $SEQ ]; then
-  (>&2 echo "ERROR: Failed to find $SEQ")
-  (>&2 echo "Please recheck the variables")
+  echo "ERROR: Failed to find $SEQ" >&2
+  echo "Please recheck the variables" >&2
   exit 1 # terminate and indicate error
 fi
 
 # File containing the encoded sequence
-ENC_FILE="enc_file.ivf"
+ENC_FILE=enc_file.ivf
 # File containing the reconstructed sequence
-REC_FILE="rec_file.y4m"
+REC_FILE=rec_file.y4m
 # File containing the decoded sequence
-DEC_FILE="dec_file.y4m"
+DEC_FILE=dec_file.y4m
 
 # Print the backtrace on error
 export RUST_BACKTRACE=1
 
 # Build and run encoder
-BUILD_TYPE=""
-if [ $IS_RELEASE == 1 ]; then
-  BUILD_TYPE="--release"
+
+if ! type cargo > /dev/null 2>&1; then
```

Comment:
This check is superfluous, because of  `set -e` we would stop at the following call with `sh: 50: cargo: not found`.

Reply:
I mainly did this to not rely on the shell's error reporting and that way if we want, we can print out a more helpful message like `cargo not found, please install rust either from your package manager or from https://rustup.rs/`, but I first wanted some feedback on this before adding links or changing error messages